components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -1081,9 +1081,11 @@ public RoleBasicInfo setPermissionsForRole(String roleID, List<String> permissio
             return new RoleBasicInfo(roleID, roleName);
         }
         try {
+            String loggedInUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
             PrivilegedCarbonContext.startTenantFlow();
             PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
             carbonContext.setTenantDomain(tenantDomain, true);
+            carbonContext.setUsername(loggedInUser);
```

Comment:
is the new tenant domain different from the previous one? if so we may not need to start the tenant flow here at all

Reply:
It could be different due to deviations of certain flows. This flow needs to be started for the required tenant to make sure these operations are carried out for the intended tenant.

Reply:
No deviations of the flow was discovered. Hence the tenant flow initiation was removed in 29de1b6baa48638a10cab8b287ff872932450233